SimpleTotp/TotpProvider.cs

```diff
@@ -0,0 +1,187 @@
+﻿// ReSharper disable StringLiteralTypo
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography;
+using System.Text;
+
+[assembly:InternalsVisibleTo("SimpleTotp.Tests")]
+
+namespace SimpleTotp
+{
+    /// <inheritdoc />
+    public class TotpProvider : ITotpProvider
+    {
+        /// <summary>
+        /// Unix Epoch time - in ticks
+        /// </summary>
+        private const long UnixEpochInTicks = 621355968000000000L;
+
+        /// <summary>
+        /// Period for which the same code should be calculated - 30 seconds (in ticks)
+        /// <remarks>
+        /// TOTP specification allows this to be customized; most existing authenticators use a static value of 30 seconds - hence a const
+        /// </remarks>
+        /// </summary>
+        private const long Period = 300000000L;
+
+        /// <summary>
+        /// Number of digits in generated TOTP code
+        /// <remarks>
+        /// TOTP specification allows this to be customized; most existing authenticators use a static value of 6 digits - hence a const
+        /// </remarks>
+        /// </summary>
+        private const int CodeDigits = 6;
+
+        /// <summary>
+        /// Default tolerance for TOTP code validation
+        /// </summary>
+        private static readonly TimeSpan DefaultTolerance = TimeSpan.FromMinutes(1);
+
+        /// <inheritdoc />
+        public RegistrationData GetAuthenticatorRegistrationData(String accountName,
+                                                                 String issuer,
+                                                                 String secretKey = null,
+                                                                 bool prefixAccountNameWithIssuer = true)
+        {
+            ValidationHelper.CheckNotNullOrWhitespace(() => issuer);
+            ValidationHelper.CheckNotNullOrWhitespace(() => accountName);
+
+            if (issuer.Contains(":"))
+                throw new ArgumentException($"{nameof(issuer)} contains a colon, which is not allowed", nameof(issuer));
```

Comment:
We use helper to check for nulls or white spaces and throw exception if it does, so why not use same logic here and use helper function?

Reply:
Mainly because that's 2 repetitions in a single method vs. multiple repetitions across several methods. I'll leave this to you if you feel it's useful to refactor. :)